code/modules/antagonists/eldritch_cult/eldritch_magic.dm

```diff
@@ -0,0 +1,395 @@
+	playsound(get_turf(user), 'sound/items/welder.ogg', 75, TRUE)
+	for(var/turf/T in targets)
+		///What we want is the 3 tiles around the user and the tile under him to be rusted, so min(dist,1)-1 causes us to get 0 for these tiles, rest of the tiles are based on chance
+		var/chance = 100 - (max(get_dist(T,user),1)-1)*100/(range+1)
+		if(!prob(chance))
+			continue
+		if(!is_type_in_typecache(T, blacklisted_turfs))
+			T.ChangeTurf(/turf/open/floor/plating/rust)
+			continue
+		if(T.type == /turf/closed/wall/rust)
+			T.ScrapeAway()
+			continue
+		if(T.type == /turf/closed/wall/r_wall/rust && prob(50))
+			T.ScrapeAway()
+			continue
+		if(T.type == /turf/closed/wall)
+			T.ChangeTurf(/turf/closed/wall/rust)
+			continue
+		if(T.type == /turf/closed/wall/r_wall && prob(50))
+			T.ChangeTurf(/turf/closed/wall/r_wall/rust)
+			continue
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/small
+	name = "Rust Conversion"
+	desc = "Spreads rust onto nearby turfs."
+	range = 2
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/big
+	name = "Rustbringer's rite"
+	desc = "Spreads rust onto a lot of turfs, and damages items worn on other peoplke"
+	range = 8
+
+/obj/effect/proc_holder/spell/aoe_turf/rust_conversion/big/cast(list/targets, mob/user)
+	. = ..()
+	for(var/turf/T in targets)
+
+		for(var/Y in T.GetAllContents())
+
+			if(!is_type_in_list(/mob/living/carbon,T.GetAllContents()))
+				break
+
+			if(!istype(Y,/mob/living/carbon))
+				continue
+
+			var/mob/living/carbon/C = Y
+			if(C == user)
+				continue
+
+			for(var/X in C.GetAllContents())
+				if(istype(X,/obj/item))
+					var/obj/item/I = X
+					if(prob(75))
+						I.take_damage(rand(50,500))
+
+/obj/effect/proc_holder/spell/targeted/touch/ash_leech
+	name = "Blood Siphon"
+	desc = "Touch spell that heals you while damaging the enemy."
+	hand_path = /obj/item/melee/touch_attack/ash_leech
+	school = "evocation"
+	charge_max = 150
+	clothes_req = FALSE
+	invocation = "FL'MS O'ET'RN'ITY"
+	invocation_type = "whisper"
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "ash_shift"
+	action_background_icon_state = "bg_ecult"
+
+/obj/item/melee/touch_attack/ash_leech
+	name = "Blood Siphon"
+	desc = "A sinister looking aura that distorts the flow of reality around it."
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	catchphrase = "R'BRTH"
+
+/obj/item/melee/touch_attack/ash_leech/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	..()
+	if(ishuman(target))
+		var/mob/living/carbon/human/tar = target
+		if(tar.anti_magic_check())
+			tar.visible_message("<span class='danger'>Spell bounces off of [target]!</span>","<span class='danger'>The spell bounces off of you!</span>")
+			return
+	if(iscarbon(target))
+		var/mob/living/carbon/C1 = target
+		var/mob/living/carbon/C2 = user
+		C1.adjustBruteLoss(20)
+		C1.blood_volume -= 100
+		C2.adjustBruteLoss(-20)
+		C2.blood_volume += 100
```

Comment:
if you cast this spell enough, I think this might be able to take you above the blood level threshold that gibs you if you go over it

I fucking told you people that that PR was a bad idea

Reply:
i will add acheck for it